lib/darr.h

```diff
@@ -102,7 +102,7 @@ void *_darr__resize(void *a, uint count, size_t esize, struct memtype *mt);
 
 #define _darr_esize(A) sizeof((A)[0])
 #define _darr_len(A)   _darr_meta(A)->len
-#define _darr_meta(A)  (((struct darr_metadata *)(A)) - 1)
+#define _darr_meta(A)  ((struct darr_metadata *)(((intptr_t)(A)) - sizeof(struct darr_metadata)))
```

Comment:
So, as I'm sure you agree, this code is actually much less clear and much more error prone that what it replaces. What is the advantage here?

Reply:
I absolutely agree the code is less clear.

However, I also absolutely disagree with disabling this specific warning, because it occurs exactly this once, and it is quite useful in other contexts. (The warning is "out of bounds", not "negative index".)

Why GCC is warning for this one specific location, and nowhere else, I have no idea. I've tried poking at it and haven't been able to figure out any specific reason. The `darr` code is used in other places and none of those trigger warnings.

Is it possible there's an actual bug in this usage location? I haven't dug through to see if  something is different there, or maybe it's using a `darr` function not used elsewhere?

Reply:
If anything, the alternate workaround would be sticking `_Pragma("GCC warning disable yadda yadda")` in the macro, which is also ugly as hell but would at least allow the code to remain untouched.